poollet/machinepoollet/controllers/machine_controller.go

```diff
@@ -509,6 +510,10 @@ func (r *MachineReconciler) updateNetworkInterfaceStatus(
 				errs = append(errs, err)
 			}
 		}
+
+		if err := r.updateNetworkInterfaceStatusFromIRIStatus(ctx, nic, iriNicStatus); err != nil {
+			errs = append(errs, err)
+		}
 	}
 
 	for _, nic := range unhandledNicByUID {
```

Comment:
Do we here also need `r.updateNetworkInterfaceStatusFromIRIStatus` ?

Reply:
Well in that case I don't have a corresponding IRI network interface, hence can't really calculate any state/ips for status. The default `Pending` state is the way to go here.